### DISCUSSION TITLE
How to know in Lua that `--embed-resources` is being used?

### DISCUSSION THREAD

**Discussion by mcanouil** [jgm/pandoc#9978] [Q&A]
Is there a way in Lua to know that `--embed-resources` is being used?
It does not seem to be inside any Pandoc variables.

**Comment by tarleb** [comment]
You're right, this information is not available in Lua. Embedding is done in the main pandoc app, and all information about that is contained there.

**Reply by mcanouil** [reply]
Would it be possible to add that information? (I can make a more formal feature request)

**Reply by tarleb** [reply]
I see two possible approaches:

1. Move the embedding operation from the main app to the HTML writer, the information would then become available via the writer options.
2. Make the full set of command line options part of the "filter environment", i.e., the information that's passed to filters.

The second option isn't very clean and rather involved, so I guess there's only one reasonable approach. I haven't thought that one through, not sure if that would bring any problems.

**Reply by mcanouil** [reply]
So not a trivial change to make in Pandoc.

**Reply by bpj** [reply]
I would typically pass a metadata variable manually to trigger whatever different behavior I want. Both clunky and reliable at the same time...

**Reply by tarleb** [reply]
I checked, it seems to be a rather simple change. Proof of Concept: #9982
